setup.py

```diff
@@ -83,24 +83,28 @@ def run(self):
 
     def build_extension(self, ext):
         extdir = os.path.abspath(os.path.dirname(self.get_ext_fullpath(ext.name)))
+
+        build_type =  os.environ.get("SONATA_BUILD_TYPE", "Release")
+        if self.debug:
+            build_type = "Debug"
+
         cmake_args = [
             "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=" + extdir,
             "-DSONATA_TESTS={}".format(os.environ.get("SONATA_TESTS", "OFF")),
             "-DEXTLIB_FROM_SUBMODULES=ON",
             "-DSONATA_PYTHON=ON",
             "-DSONATA_VERSION=" + self.distribution.get_version(),
-            "-DCMAKE_BUILD_TYPE=",
+            "-DCMAKE_BUILD_TYPE={}".format(build_type),
             "-DSONATA_CXX_WARNINGS=OFF",
             '-DPYTHON_EXECUTABLE=' + sys.executable
         ]
 
-        optimize = "OFF" if self.debug else "ON"
-        build_args = ["--config", optimize, "--target", self.target]
+        build_args = ["--config", build_type, "--target", self.target]
 
         if platform.system() == "Windows":
             cmake_args += [
                 "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_{}={}".format(
-                    optimize.upper(), extdir
+                    build_args.upper(), extdir
```

Comment:
uh, `build_args` is a list?
```suggestion
                    build_type.upper(), extdir
```

Reply:
typo-fixed - tnx